pkg/request/pending.go

```diff
@@ -23,10 +27,10 @@ type PendingStore struct {
 	ReqIDGCInterval       time.Duration
 	ReqIDLifetime         time.Duration
 	Logger                Logger
-	Inspector             RequestInspector
+	Inspector             api.RequestInspector
```

Comment:
no, please let's not couple this component with the `pkg/` types. 

We can just define them here, as needed.